vpr/src/place/initial_placement.cpp

```diff
@@ -1892,13 +1893,17 @@ static inline void place_all_blocks_ap(enum e_pad_loc_type pad_loc_type,
 
     if (constrained_clusters.size() > 0) {
         VTR_LOG("Placing constrained clusters...\n");
-        place_blocks_min_displacement(constrained_clusters,
+        bool all_clusters_placed = place_blocks_min_displacement(constrained_clusters,
                                       pad_loc_type,
                                       blk_loc_registry,
                                       place_macros,
                                       cluster_netlist,
                                       flat_placement_info);
         VTR_LOG("\n");
+        if (!all_clusters_placed) {
+            VTR_LOG("Could not place all constrained clusters, falling back on original IP.\n");
```

Comment:
Another nitpick: I think it’s better to avoid abbreviations in logs whenever possible. I suggest changing “IP” to “initial placement.”